CMake/FollyCompilerUnix.cmake

```diff
@@ -2,7 +2,6 @@ set(CMAKE_CXX_FLAGS_COMMON "-g -Wall -Wextra")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_COMMON}")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_COMMON} -O3")
 
-list(APPEND CMAKE_REQUIRED_FLAGS -std=gnu++14)
```

Comment:
It looks like removing this breaks a number of the compilation checks on Linux.
For instance, the FOLLY_HAVE_WEAK_SYMBOLS and FOLLY_HAVE_LINUX_VDSO checks incorrectly fail with this change.  I haven't looked too closely but I suspect they are failing because they use `nullptr` in the checks, and require C++ 11 or greater functionality to compile.

Reply:
Whoops... Apparently I wasn't reading `CMakeError.log` closely enough, because those tests fail for the same reason on my machine.

The other option I was playing with when writing this PR was using `string(REGEX REPLACE)` right before the `check_symbol_exists` call to strip out C++ version flags. I thought that was pretty fragile, though, so I abandoned that approach. Do you think it's worth another shot? Or got any better ideas?

Also, for cases like this, do you want me to use a regular revert + new commit, or rewrite things so the bad commit never happened?

Reply:
All of the commits will get squashed, and the commit message will be the message put in for the Pull Request, so the individual commits don't actually matter.